rustfmt.toml

```
@@ -0,0 +1,6 @@
+hard_tabs = true
+use_small_heuristics = "Max"
+max_width = 100
+chain_width = 80
+spaces_around_ranges = true
+binop_separator = "Back"
```

```suggestion
hard_tabs = true
use_small_heuristics = "Max"
max_width = 100
comment_width = 100
chain_width = 100
merge_imports = true
reorder_imports = false
spaces_around_ranges = true
binop_separator = "Back"
```

This is what I recommend, some of them are unstable though.

---

> reorder_imports = false

Why not true?

---

Merging imports is a good idea. I think we should reorder them though, as they're unordered at the moment just because nobody (rightfully) can be bothered to do that manually. I think `chain_width` should be less than `max_width` as long chains are quite hard to read.

---

I wish there was an option to order in groups, so std/core, crates.io, crates same repo, same crate.

---

> I wish there was an option to order in groups, so std/core, crates.io, crates same repo, same crate.

Yeah, it's a shame all the `sp_` imports get mixed in with the rest.

---

Added https://github.com/rust-lang/rustfmt/issues/4860

---

The reordering of imports does not cross empty lines so you can still group std and others for example.

---

I think expenses has integrated all these suggestions.